.github/workflows/main.yml

```diff
@@ -1,14 +1,6 @@
 name: Build and Test
 
-on:
-  push:
-    branches:
-    - 'master'
-    tags:
-    - 'v*'
-  pull_request:
-    branches:
-    - '*'
+on: [push, pull_request]
```

Comment:
but then you would upload any binary on pushes to branches != master. This doesn't seem right

Reply:
Artifact upload will happen for all cases, artifacts are something that CI will produce but will only be available inside the CI UI: Actions -> specific build -> artifact will be there.

For github releases, it'll depends of what kind of push was done, if it's tag, an official binary will be available, otherwise a binary with "prerelease" mark will be created.
`if: ${{ github.event_name == 'push' }}`, `prerelease: ${{ !startsWith(github.ref, 'refs/tags/') }}`
I can remove prerelease binaries if that's something problematic.

Reply:
ping @lucasdemarchi

Reply:
artifacts take space and are eventually blocked. I don't know how much space it is, but I hit the max in another internal project. I'd prefer that we only upload artifacts for master branch and for pull requests, but not for random branches

Reply:
@lucasdemarchi done.